new ppc64 profiles work in progress

gyakovlev opened the pull request:
# new profiles for ppc64 and ppc as well
Why?
because current design is old and hard to follow, talks about dropping some profiles started in 2017 and cleanup is long overdue.

## Proposal
1) split out profiles from `arch/powerpc` common directory and move to `arch/ppc` and `arch/ppc64` accordingly.
2) introduce endian specific arch subdirs on `ppc64`, e.g. `arch/ppc64/little-endian` and `arch/ppc64/big-endian`. all common settings should be in `arch/ppc64` and endian-specific profiles should contain `make.default` with profile specific triples and masks.
3) create new user-visible (`eselect profile list`) profiles, proposed names are:

    ` default/linux/ppc64/17.0 .. {musl,hardened,systemd,dekstop etc)`
    ` default/linux/ppc64le/17.0`
     `default/linux/ppc/17.0`

and parent via `arch/ppc{,64}`

4) users who wish to use 32-bit userland can just switch to `ppc` profile and specify `ABI_KERNEL=ppc64`, gcc it multiarch and should handle `-m64` on 32bit userland. need to make sure it's visible.
5) finally introduce desktop subprofile for `ppc64le`,  also hardened and musl profiles (TODO) for rest, stage already created and we can get it rolling.



## The rollout plan
1) introduce new profiles which inherit new `arch/ppc64` structure`
2) create news item
3) depcrecate profile, no migration required for ppc64 users. ppc64/32ul patch TBD.
4) wait, and disable.
5) after migration period is over, dissolve current `arch/powerpc` subdir into `arch/ppc` and `arch/ppc64`, splitting profiles.


developers should not touch `arch/ppc{,64}` profiles for masks, and continue using `arch/powerpc` before migration is done.


to compare resulting profiles I'm using
https://github.com/mgorny/profile-cleanup/blob/master/profile-dumper

current results are quite promising, comparing existing with new profiles.
```
diff -r old new
45c45
< PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
---
> PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
50d49
< SYMLINK_LIB="no"
```

as you can see only thing I lost is SYMLINK_LIB
rest is just script being stupid and not de-duping variables.

it's defined here
```
ppc64/17.0/64bit-userland/make.defaults
5:SYMLINK_LIB="yes"

ppc64/17.0/64bit-userland/little-endian/make.defaults
5:SYMLINK_LIB="no"
```

but It's a good chance to kill it.

## big endian
### default/linux/ppc64/17.0
![ppc64-17 0](https://user-images.githubusercontent.com/168902/72957031-f355d280-3d99-11ea-89fb-0b93adb2d2ed.png)

### old profile default/linux/powerpc/ppc64/17.0/64bit-userland
![old-ppc64-17 0](https://user-images.githubusercontent.com/168902/72957039-f8b31d00-3d99-11ea-94d8-6f07557afb5a.png)


## little endian
### default/linux/ppc64le/17.0
![ppc64le-17 0](https://user-images.githubusercontent.com/168902/72957044-0072c180-3d9a-11ea-8af7-67aba41d5ca0.png)
### default/linux/powerpc/ppc64/17.0/64bit-userland/little-endian
![old-ppc64le-17 0](https://user-images.githubusercontent.com/168902/72957047-023c8500-3d9a-11ea-940d-452d4c3c73d3.png)



after proper flattening and removal of `arch/powerpc` profiles will become even flatter and cleaner and easier to follow.

gyakovlev commented:
@gentoo/ppc ^

blueness commented:
This is incomplete.  The current 32-bit profiles also have uclibc and musl profiles.  These must be included.

To be specific, here's what `eselect profile list` gives on 32-bit:

```
  [1]   default/linux/powerpc/ppc32/17.0 (stable) *
  [2]   default/linux/powerpc/ppc32/17.0/desktop (stable)
  [3]   default/linux/powerpc/ppc32/17.0/desktop/gnome (stable)
  [4]   default/linux/powerpc/ppc32/17.0/desktop/gnome/systemd (stable)
  [5]   default/linux/powerpc/ppc32/17.0/developer (stable)
  [6]   default/linux/powerpc/ppc64/17.0/32bit-userland (stable)
  [7]   default/linux/powerpc/ppc64/17.0/32bit-userland/desktop (stable)
  [8]   default/linux/powerpc/ppc64/17.0/32bit-userland/desktop/gnome (stable)
  [9]   default/linux/powerpc/ppc64/17.0/32bit-userland/desktop/gnome/systemd (stable)
  [10]  default/linux/powerpc/ppc64/17.0/32bit-userland/developer (stable)
  [11]  default/linux/powerpc/ppc32/17.0/musl (exp)
  [12]  default/linux/powerpc/ppc32/17.0/musl/hardened (exp)
  [13]  default/linux/powerpc/ppc32/17.0/uclibc (exp)
  [14]  default/linux/powerpc/ppc32/17.0/uclibc/hardened (exp)
```
I see nothing in your scheme for profiles 11 through 14.

gyakovlev commented:
@blueness of course I’ll add those profiles, i just wanted to get some input on top level layout before I start pushing forward and stacking on top of this structure.

gyakovlev commented:
pushed ppc subdir and top level profiles,
now going to go deeper and re-implement all feature profiles.


### default/linux/powerpc/ppc32/17.0
![old-ppc32-17 0](https://user-images.githubusercontent.com/168902/73115307-17412180-3f1c-11ea-8b78-947efa6326ed.png)

compared to 
### default/linux/powerpc/ppc32/17.0
![ppc-17 0](https://user-images.githubusercontent.com/168902/73115309-1ad4a880-3f1c-11ea-96b6-d51bb5674389.png)

gyakovlev commented:
new profile in order

![ppc-17 0-desktop](https://user-images.githubusercontent.com/168902/73115656-f0d1b500-3f20-11ea-90a8-d55c3cf313d4.png)
![ppc-17 0-desktop-gnome](https://user-images.githubusercontent.com/168902/73115657-f4653c00-3f20-11ea-8b71-8fceb27a1fbd.png)
![ppc-17 0-desktop-gnome-systemd](https://user-images.githubusercontent.com/168902/73115660-f929f000-3f20-11ea-8079-7a77a649ff54.png)
![ppc-17 0-developer](https://user-images.githubusercontent.com/168902/73115661-faf3b380-3f20-11ea-8f1c-42ae2677ec8a.png)

gyakovlev commented:
now the most insteresting stuff, musl profile for `ppc`

due to way profiles were stacked and triple referenced it was required to patent to arch again before musl feature

https://github.com/gentoo/gentoo/blob/261f747b27a57db8c9da8fc9f2d1a991a5db350e/profiles/default/linux/powerpc/ppc32/17.0/musl/parent#L1-L3

with new profiles it's cleaner and no need for hacks.

![ppc-17 0-musl](https://user-images.githubusercontent.com/168902/73115799-12cc3700-3f23-11ea-9f22-bd61902fbdb6.png)



old one...
![old-ppc32-17 0-musl](https://user-images.githubusercontent.com/168902/73115805-3db68b00-3f23-11ea-80ce-00882be09355.png)

gyakovlev commented:
I discovered that old ppc32 musl hardened profile was not actually hardened, as it did not parent to hardened feature at all.
![old-ppc32-17 0-musl-hardened](https://user-images.githubusercontent.com/168902/73115944-36dd4780-3f26-11ea-8d60-bccee9822210.png)


new one does it properly
![ppc-17 0-musl-hardened](https://user-images.githubusercontent.com/168902/73115946-3fce1900-3f26-11ea-854b-5794120b793a.png)

gyakovlev commented:
another discovery, uclibc/hardened profile is not only not hardened, it actually parents to musl feature
probably copy/paste err
![old-ppc32-17 0-uclibc-hardened-WTHISISWRONG](https://user-images.githubusercontent.com/168902/73116064-2928c180-3f28-11ea-930e-8a52ffe20d9a.png)


new  uclibc profiles are ok
![ppc-17 0-uclibc](https://user-images.githubusercontent.com/168902/73116069-3f368200-3f28-11ea-90a4-28620fa3148b.png)
![ppc-17 0-uclibc-hardened](https://user-images.githubusercontent.com/168902/73116070-42317280-3f28-11ea-9d05-e2ebd6168c8d.png)

gyakovlev commented:
and last bunch of ppc64 profiles
*
![ppc64-17 0-desktop](https://user-images.githubusercontent.com/168902/73116376-241a4100-3f2d-11ea-855d-0219d75a77a6.png)
*
![ppc64-17 0-desktop-gnome](https://user-images.githubusercontent.com/168902/73116377-267c9b00-3f2d-11ea-8bab-29880818bd3f.png)
*
![ppc64-17 0-desktop-gnome-systemd](https://user-images.githubusercontent.com/168902/73116378-2aa8b880-3f2d-11ea-8eda-60236f86a5cd.png)
*
![ppc64le-17 0-systemd](https://user-images.githubusercontent.com/168902/73116381-2f6d6c80-3f2d-11ea-88c3-642180e7dd30.png)

gyakovlev commented:
and just to clarify, musl profiles for ppc64 also planned.
but to do that we'll ned to restore ppc64 keywords in musl overlay.

seed stages available at https://dev.gentoo.org/~gyakovlev/stages/
those are not proper stages, but should be enough to re-bootstrap stage with new profile and do a catalyst run.

gyakovlev commented:
I'm now writing automated test to verify all profiles, will post later.

gyakovlev commented:
current test results

looks mostly ok but still need some work ofc.
due to multiple inheritance some variables are tripled in `PROFILE_ONLY_VARIABLES` and it's ok to ignore.

musl and uclibc hardened profiles are broken, it's expected, see 
https://github.com/gentoo/gentoo/pull/14413#issuecomment-578373961
https://github.com/gentoo/gentoo/pull/14413#issuecomment-578374720

```diff
####################
default/linux/powerpc/ppc32/17.0 is now default/linux/ppc/17.0
diff -u default_linux_powerpc_ppc32_17.0/make.defaults default_linux_ppc_17.0/make.defaults
--- default_linux_powerpc_ppc32_17.0/make.defaults	2020-01-24 21:40:36.696945326 -0800
+++ default_linux_ppc_17.0/make.defaults	2020-01-24 21:40:36.806933351 -0800
@@ -36,7 +36,7 @@
 OFFICE_IMPLEMENTATION="libreoffice"
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0/package.use.mask default_linux_ppc_17.0/package.use.mask
--- default_linux_powerpc_ppc32_17.0/package.use.mask	2020-01-24 21:40:36.700278296 -0800
+++ default_linux_ppc_17.0/package.use.mask	2020-01-24 21:40:36.813599292 -0800
@@ -207,7 +207,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/desktop is now default/linux/ppc/17.0/desktop
diff -u default_linux_powerpc_ppc32_17.0_desktop/make.defaults default_linux_ppc_17.0_desktop/make.defaults
--- default_linux_powerpc_ppc32_17.0_desktop/make.defaults	2020-01-24 21:40:36.933586230 -0800
+++ default_linux_ppc_17.0_desktop/make.defaults	2020-01-24 21:40:37.046907226 -0800
@@ -36,7 +36,7 @@
 OFFICE_IMPLEMENTATION="libreoffice"
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_desktop/package.use.mask default_linux_ppc_17.0_desktop/package.use.mask
--- default_linux_powerpc_ppc32_17.0_desktop/package.use.mask	2020-01-24 21:40:36.940252171 -0800
+++ default_linux_ppc_17.0_desktop/package.use.mask	2020-01-24 21:40:37.053573167 -0800
@@ -218,7 +218,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/desktop/gnome is now default/linux/ppc/17.0/desktop/gnome
diff -u default_linux_powerpc_ppc32_17.0_desktop_gnome/make.defaults default_linux_ppc_17.0_desktop_gnome/make.defaults
--- default_linux_powerpc_ppc32_17.0_desktop_gnome/make.defaults	2020-01-24 21:40:37.176893076 -0800
+++ default_linux_ppc_17.0_desktop_gnome/make.defaults	2020-01-24 21:40:37.296880013 -0800
@@ -36,7 +36,7 @@
 OFFICE_IMPLEMENTATION="libreoffice"
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_desktop_gnome/package.use.mask default_linux_ppc_17.0_desktop_gnome/package.use.mask
--- default_linux_powerpc_ppc32_17.0_desktop_gnome/package.use.mask	2020-01-24 21:40:37.183559016 -0800
+++ default_linux_ppc_17.0_desktop_gnome/package.use.mask	2020-01-24 21:40:37.300212983 -0800
@@ -218,7 +218,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/desktop/gnome/systemd is now default/linux/ppc/17.0/desktop/gnome/systemd
diff -u default_linux_powerpc_ppc32_17.0_desktop_gnome_systemd/make.defaults default_linux_ppc_17.0_desktop_gnome_systemd/make.defaults
--- default_linux_powerpc_ppc32_17.0_desktop_gnome_systemd/make.defaults	2020-01-24 21:40:37.436864773 -0800
+++ default_linux_ppc_17.0_desktop_gnome_systemd/make.defaults	2020-01-24 21:40:37.560184681 -0800
@@ -36,7 +36,7 @@
 OFFICE_IMPLEMENTATION="libreoffice"
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_desktop_gnome_systemd/package.use.mask default_linux_ppc_17.0_desktop_gnome_systemd/package.use.mask
--- default_linux_powerpc_ppc32_17.0_desktop_gnome_systemd/package.use.mask	2020-01-24 21:40:37.443530714 -0800
+++ default_linux_ppc_17.0_desktop_gnome_systemd/package.use.mask	2020-01-24 21:40:37.566850622 -0800
@@ -220,7 +220,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/developer is now default/linux/ppc/17.0/developer
diff -u default_linux_powerpc_ppc32_17.0_developer/make.defaults default_linux_ppc_17.0_developer/make.defaults
--- default_linux_powerpc_ppc32_17.0_developer/make.defaults	2020-01-24 21:40:37.690170530 -0800
+++ default_linux_ppc_17.0_developer/make.defaults	2020-01-24 21:40:37.810157468 -0800
@@ -39,7 +39,7 @@
 PHP_TARGETS="php7-2"
 PORTAGE_ELOG_CLASSES=" qa"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_developer/package.use.mask default_linux_ppc_17.0_developer/package.use.mask
--- default_linux_powerpc_ppc32_17.0_developer/package.use.mask	2020-01-24 21:40:37.696836471 -0800
+++ default_linux_ppc_17.0_developer/package.use.mask	2020-01-24 21:40:37.813490438 -0800
@@ -218,7 +218,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/musl is now default/linux/ppc/17.0/musl
diff -u default_linux_powerpc_ppc32_17.0_musl/make.defaults default_linux_ppc_17.0_musl/make.defaults
--- default_linux_powerpc_ppc32_17.0_musl/make.defaults	2020-01-24 21:40:37.946809258 -0800
+++ default_linux_ppc_17.0_musl/make.defaults	2020-01-24 21:40:38.066796195 -0800
@@ -39,7 +39,7 @@
 PHP_TARGETS="php7-2"
 PORTAGE_LIBC="musl"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_musl/package.use.mask default_linux_ppc_17.0_musl/package.use.mask
--- default_linux_powerpc_ppc32_17.0_musl/package.use.mask	2020-01-24 21:40:37.956808169 -0800
+++ default_linux_ppc_17.0_musl/package.use.mask	2020-01-24 21:40:38.076795107 -0800
@@ -208,7 +208,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/musl/hardened is now default/linux/ppc/17.0/musl/hardened
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/make.defaults default_linux_ppc_17.0_musl_hardened/make.defaults
--- default_linux_powerpc_ppc32_17.0_musl_hardened/make.defaults	2020-01-24 21:40:38.210113926 -0800
+++ default_linux_ppc_17.0_musl_hardened/make.defaults	2020-01-24 21:40:38.333433834 -0800
@@ -6,7 +6,7 @@
 ALSA_CARDS="aoa aoa-fabric-layout aoa-onyx aoa-soundbus aoa-soundbus-i2s aoa-tas aoa-toonie powermac usb-audio via82xx"
 APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias"
 ARCH="ppc"
-BOOTSTRAP_USE="unicode internal-glib pkg-config split-usr python_targets_python3_6 python_targets_python2_7 nptl -berkdb -gdbm nptl -berkdb -gdbm"
+BOOTSTRAP_USE="unicode internal-glib pkg-config split-usr python_targets_python3_6 python_targets_python2_7 nptl -berkdb -gdbm hardened pic xtpax -jit -orc"
 CALLIGRA_FEATURES="karbon sheets words"
 CFLAGS="-O2 -pipe"
 CHOST="powerpc-gentoo-linux-musl"
@@ -36,10 +36,12 @@
 MULTILIB_ABIS="ppc"
 NETBEANS_MODULES="apisupport cnd groovy gsf harness ide identity j2ee java mobility nb php profiler soa visualweb webcommon websvccommon xml"
 OFFICE_IMPLEMENTATION="libreoffice"
+PAX_MARKINGS="none"
 PHP_TARGETS="php7-2"
 PORTAGE_LIBC="musl"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_IS_HARDENED="1"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
@@ -47,7 +49,7 @@
 SYMLINK_LIB="no"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
-USE="acl -berkdb bzip2 cli crypt cxx dri fortran -gdbm iconv ipv6 ncurses nls nptl openmp pam pcre readline seccomp split-usr ssl tcpd unicode xattr zlib"
+USE="acl -berkdb bzip2 -cli crypt cxx -dri -fortran -gdbm hardened iconv ipv6 -jit ncurses nls nptl openmp -orc pam pcre pic -ptpax readline seccomp split-usr ssl -tcpd unicode xattr xtpax zlib"
 USERLAND="GNU"
 USE_EXPAND="ABI_MIPS ABI_PPC ABI_RISCV ABI_S390 ABI_X86 ADA_TARGET ALSA_CARDS APACHE2_MODULES APACHE2_MPMS CALLIGRA_FEATURES CAMERAS COLLECTD_PLUGINS CPU_FLAGS_ARM CPU_FLAGS_PPC CPU_FLAGS_X86 CURL_SSL ELIBC ENLIGHTENMENT_MODULES FFTOOLS GPSD_PROTOCOLS GRUB_PLATFORMS INPUT_DEVICES KERNEL L10N LCD_DEVICES LIBREOFFICE_EXTENSIONS LIRC_DEVICES LLVM_TARGETS MONKEYD_PLUGINS NETBEANS_MODULES NGINX_MODULES_HTTP NGINX_MODULES_MAIL NGINX_MODULES_STREAM OFED_DRIVERS OFFICE_IMPLEMENTATION OPENMPI_FABRICS OPENMPI_OFED_FEATURES OPENMPI_RM PHP_TARGETS POSTGRES_TARGETS PYTHON_SINGLE_TARGET PYTHON_TARGETS QEMU_SOFTMMU_TARGETS QEMU_USER_TARGETS ROS_MESSAGES RUBY_TARGETS SANE_BACKENDS USERLAND UWSGI_PLUGINS VIDEO_CARDS VOICEMAIL_STORAGE XFCE_PLUGINS XTABLES_ADDONS"
 USE_EXPAND_HIDDEN="ABI_MIPS ABI_PPC ABI_RISCV ABI_S390 ABI_X86 CPU_FLAGS_ARM -CPU_FLAGS_PPC CPU_FLAGS_X86 ELIBC KERNEL USERLAND"
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/package.mask default_linux_ppc_17.0_musl_hardened/package.mask
--- default_linux_powerpc_ppc32_17.0_musl_hardened/package.mask	2020-01-24 21:40:38.223445808 -0800
+++ default_linux_ppc_17.0_musl_hardened/package.mask	2020-01-24 21:40:38.346765716 -0800
@@ -2,12 +2,16 @@
 <dev-util/nvidia-cuda-sdk-9
 <dev-util/nvidia-cuda-toolkit-9
 <sys-devel/gcc-6
+<sys-libs/glibc-2.25
 =virtual/libudev-232-r1
 >=www-client/firefox-54.0
+app-crypt/hmaccalc
 dev-java/ecj-gcj
 dev-java/gcj-jdk
 dev-lang/rust-bin
+net-fs/openafs-kernel
 sys-apps/systemd
+sys-devel/prelink
 sys-fs/udev
 sys-libs/glibc
 sys-libs/uclibc-ng
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/packages default_linux_ppc_17.0_musl_hardened/packages
--- default_linux_powerpc_ppc32_17.0_musl_hardened/packages	2020-01-24 21:40:38.210113926 -0800
+++ default_linux_ppc_17.0_musl_hardened/packages	2020-01-24 21:40:38.333433834 -0800
@@ -13,6 +13,7 @@
 *sys-apps/busybox
 *sys-apps/coreutils
 *sys-apps/diffutils
+*sys-apps/elfix
 *sys-apps/file
 *sys-apps/gawk
 *sys-apps/grep
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/package.use default_linux_ppc_17.0_musl_hardened/package.use
--- default_linux_powerpc_ppc32_17.0_musl_hardened/package.use	2020-01-24 21:40:38.216779867 -0800
+++ default_linux_ppc_17.0_musl_hardened/package.use	2020-01-24 21:40:38.333433834 -0800
@@ -1,6 +1,7 @@
 =dev-lang/python-2* -berkdb
 app-admin/sudo -ldap
 app-emulation/qemu qemu_softmmu_targets_ppc
+dev-lang/python gdbm
 dev-ml/llvm-ocaml llvm_targets_BPF
 dev-vcs/git-bz python_single_target_python2_7 python_targets_python2_7
 gnome-base/libglade python_single_target_python2_7 python_targets_python2_7
@@ -11,5 +12,6 @@
 sys-boot/grub:2 grub_platforms_ieee1275
 sys-devel/clang llvm_targets_BPF
 sys-devel/llvm llvm_targets_BPF
+sys-libs/gdbm berkdb
 sys-libs/pam -filecaps
 x11-base/xorg-server xorg
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/package.use.force default_linux_ppc_17.0_musl_hardened/package.use.force
--- default_linux_powerpc_ppc32_17.0_musl_hardened/package.use.force	2020-01-24 21:40:38.216779867 -0800
+++ default_linux_ppc_17.0_musl_hardened/package.use.force	2020-01-24 21:40:38.333433834 -0800
@@ -4,6 +4,7 @@
 >=www-client/firefox-45.0 system-libevent
 >=www-client/seamonkey-2.40 system-libevent
 >gnome-base/nautilus-3.29 seccomp
+app-arch/tar xattr
 app-emulation/wine-staging staging
 app-misc/tracker-miners seccomp
 app-shells/bash:0 readline
@@ -25,7 +26,8 @@
 net-libs/webkit-gtk seccomp
 net-misc/netkit-bootparamd libtirpc
 sci-astronomy/esomidas libtirpc
-sys-apps/portage ipc
+sys-apps/coreutils xattr
+sys-apps/portage ipc xattr
 sys-boot/refind gnuefi
 sys-cluster/glusterfs libtirpc
 sys-devel/clang llvm_targets_PowerPC
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/package.use.mask default_linux_ppc_17.0_musl_hardened/package.use.mask
--- default_linux_powerpc_ppc32_17.0_musl_hardened/package.use.mask	2020-01-24 21:40:38.220112837 -0800
+++ default_linux_ppc_17.0_musl_hardened/package.use.mask	2020-01-24 21:40:38.336766804 -0800
@@ -137,7 +137,7 @@
 dev-util/catalyst system-bootloader
 dev-util/geany-plugins git
 dev-util/intel-ocl-sdk system-clang
-dev-vcs/git mediawiki mediawiki-experimental
+dev-vcs/git mediawiki mediawiki-experimental pcre-jit
 dev-vcs/subversion kwallet
 games-arcade/savagewheels fmod
 games-board/stockfish armv7
@@ -198,6 +198,7 @@
 net-analyzer/wireshark bcg729
 net-dns/pdnsd -isdn -urandom
 net-dns/unbound dnstap
+net-fs/openafs modules
 net-fs/samba ceph dmapi fam system-heimdal
 net-im/choqok telepathy
 net-im/pidgin zephyr
@@ -208,7 +209,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
@@ -255,7 +255,7 @@
 sys-cluster/openmpi openmpi_fabrics_dapl openmpi_fabrics_knem openmpi_fabrics_ofed openmpi_fabrics_psm openmpi_ofed_features_connectx-xrc openmpi_ofed_features_control-hdr-padding openmpi_ofed_features_dynamic-sl openmpi_ofed_features_failover openmpi_ofed_features_rdmacm openmpi_ofed_features_udcm openmpi_rm_slurm
 sys-devel/clang default-libcxx
 sys-devel/clang-runtime crt libcxx openmp sanitize
-sys-devel/gcc awt hardened jit libssp -pie sanitize vtv
+sys-devel/gcc awt -hardened jit libssp -pie sanitize vtv
 sys-devel/gettext nls
 sys-devel/llvm gold
 sys-fabric/ofed ofed_drivers_ehca
@@ -265,7 +265,7 @@
 sys-fs/multipath-tools rbd
 sys-fs/zfs kernel-builtin
 sys-kernel/linux-firmware initramfs unknown-license
-sys-libs/glibc cet hardened vanilla
+sys-libs/glibc cet -hardened vanilla
 sys-libs/gwenhywfar fox
 sys-libs/libcxx libcxxrt
 sys-libs/libxcrypt system
@@ -275,7 +275,7 @@
 virtual/mysql embedded
 virtual/notification-daemon kde
 www-client/elinks guile
-www-client/firefox lto pgo rust system-cairo
+www-client/firefox gmp-autoupdate lto pgo rust system-cairo
 www-client/links suid
 www-client/otter webengine
 www-servers/nginx luajit nginx_modules_http_lua
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/use.force default_linux_ppc_17.0_musl_hardened/use.force
--- default_linux_powerpc_ppc32_17.0_musl_hardened/use.force	2020-01-24 21:40:38.216779867 -0800
+++ default_linux_ppc_17.0_musl_hardened/use.force	2020-01-24 21:40:38.333433834 -0800
@@ -3,6 +3,8 @@
 elibc_glibc
 elibc_musl
 kernel_linux
+pie
 ppc
 split-usr
+ssp
 userland_GNU
diff -u default_linux_powerpc_ppc32_17.0_musl_hardened/use.mask default_linux_ppc_17.0_musl_hardened/use.mask
--- default_linux_powerpc_ppc32_17.0_musl_hardened/use.mask	2020-01-24 21:40:38.216779867 -0800
+++ default_linux_ppc_17.0_musl_hardened/use.mask	2020-01-24 21:40:38.333433834 -0800
@@ -107,6 +107,7 @@
 fortune
 gcl
 gnuefi
+-hardened
 hdf
 hppa
 ia64
@@ -170,6 +171,7 @@
 oracle
 orc
 packagekit
+pch
 php_targets_php5-6
 php_targets_php7-0
 php_targets_php7-1
@@ -181,6 +183,7 @@
 prefix
 prefix-guest
 prefix-stack
+prelink
 prelude
 ps3
 python_single_target_pypy3
********************
####################
default/linux/powerpc/ppc32/17.0/uclibc is now default/linux/ppc/17.0/uclibc
diff -u default_linux_powerpc_ppc32_17.0_uclibc/make.defaults default_linux_ppc_17.0_uclibc/make.defaults
--- default_linux_powerpc_ppc32_17.0_uclibc/make.defaults	2020-01-24 21:40:38.476751565 -0800
+++ default_linux_ppc_17.0_uclibc/make.defaults	2020-01-24 21:40:38.590072561 -0800
@@ -38,7 +38,7 @@
 PHP_TARGETS="php7-2"
 PORTAGE_LIBC="uClibc"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
diff -u default_linux_powerpc_ppc32_17.0_uclibc/package.use.mask default_linux_ppc_17.0_uclibc/package.use.mask
--- default_linux_powerpc_ppc32_17.0_uclibc/package.use.mask	2020-01-24 21:40:38.480084535 -0800
+++ default_linux_ppc_17.0_uclibc/package.use.mask	2020-01-24 21:40:38.593405532 -0800
@@ -209,7 +209,6 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
********************
####################
default/linux/powerpc/ppc32/17.0/uclibc/hardened is now default/linux/ppc/17.0/uclibc/hardened
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/make.defaults default_linux_ppc_17.0_uclibc_hardened/make.defaults
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/make.defaults	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/make.defaults	2020-01-24 21:40:38.843378318 -0800
@@ -6,7 +6,7 @@
 ALSA_CARDS="aoa aoa-fabric-layout aoa-onyx aoa-soundbus aoa-soundbus-i2s aoa-tas aoa-toonie powermac usb-audio via82xx"
 APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias"
 ARCH="ppc"
-BOOTSTRAP_USE="unicode internal-glib pkg-config split-usr python_targets_python3_6 python_targets_python2_7 nptl -berkdb nptl -berkdb -gdbm"
+BOOTSTRAP_USE="unicode internal-glib pkg-config split-usr python_targets_python3_6 python_targets_python2_7 nptl -berkdb hardened pic xtpax -jit -orc"
 CALLIGRA_FEATURES="karbon sheets words"
 CFLAGS="-O2 -pipe"
 CHOST="powerpc-gentoo-linux-uclibc"
@@ -17,7 +17,7 @@
 CONFIG_PROTECT_MASK="/etc/env.d /etc/gconf"
 CXXFLAGS="-O2 -pipe"
 DEFAULT_ABI="ppc"
-ELIBC="musl"
+ELIBC="uclibc"
 ENV_UNSET="DBUS_SESSION_BUS_ADDRESS DISPLAY XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR PERL_MM_OPT PERL5LIB PERL5OPT PERL_MB_OPT PERL_CORE PERLPREFIX GOBIN"
 FCFLAGS="-O2 -pipe"
 FEATURES="-multilib-strict"
@@ -25,7 +25,6 @@
 GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx"
 GRUB_PLATFORMS=""
 INPUT_DEVICES="libinput"
-INSTALL_MASK="charset.alias"
 IUSE_IMPLICIT="abi_ppc_32 prefix prefix-guest prefix-stack"
 KERNEL="linux"
 LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text"
@@ -36,10 +35,12 @@
 MULTILIB_ABIS="ppc"
 NETBEANS_MODULES="apisupport cnd groovy gsf harness ide identity j2ee java mobility nb php profiler soa visualweb webcommon websvccommon xml"
 OFFICE_IMPLEMENTATION="libreoffice"
+PAX_MARKINGS="none"
 PHP_TARGETS="php7-2"
-PORTAGE_LIBC="musl"
+PORTAGE_LIBC="uClibc"
 POSTGRES_TARGETS="postgres10 postgres11"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_IS_HARDENED="1"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
@@ -47,7 +48,7 @@
 SYMLINK_LIB="no"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
-USE="acl -berkdb bzip2 cli crypt cxx dri fortran -gdbm iconv ipv6 ncurses nls nptl openmp pam pcre readline seccomp split-usr ssl tcpd unicode xattr zlib"
+USE="acl -berkdb bzip2 -cli crypt cxx -dri -fortran -gdbm hardened iconv ipv6 -jit ncurses nls nptl openmp -orc pam pcre pic -ptpax readline seccomp split-usr ssl -tcpd unicode xattr xtpax zlib"
 USERLAND="GNU"
 USE_EXPAND="ABI_MIPS ABI_PPC ABI_RISCV ABI_S390 ABI_X86 ADA_TARGET ALSA_CARDS APACHE2_MODULES APACHE2_MPMS CALLIGRA_FEATURES CAMERAS COLLECTD_PLUGINS CPU_FLAGS_ARM CPU_FLAGS_PPC CPU_FLAGS_X86 CURL_SSL ELIBC ENLIGHTENMENT_MODULES FFTOOLS GPSD_PROTOCOLS GRUB_PLATFORMS INPUT_DEVICES KERNEL L10N LCD_DEVICES LIBREOFFICE_EXTENSIONS LIRC_DEVICES LLVM_TARGETS MONKEYD_PLUGINS NETBEANS_MODULES NGINX_MODULES_HTTP NGINX_MODULES_MAIL NGINX_MODULES_STREAM OFED_DRIVERS OFFICE_IMPLEMENTATION OPENMPI_FABRICS OPENMPI_OFED_FEATURES OPENMPI_RM PHP_TARGETS POSTGRES_TARGETS PYTHON_SINGLE_TARGET PYTHON_TARGETS QEMU_SOFTMMU_TARGETS QEMU_USER_TARGETS ROS_MESSAGES RUBY_TARGETS SANE_BACKENDS USERLAND UWSGI_PLUGINS VIDEO_CARDS VOICEMAIL_STORAGE XFCE_PLUGINS XTABLES_ADDONS"
 USE_EXPAND_HIDDEN="ABI_MIPS ABI_PPC ABI_RISCV ABI_S390 ABI_X86 CPU_FLAGS_ARM -CPU_FLAGS_PPC CPU_FLAGS_X86 ELIBC KERNEL USERLAND"
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.mask default_linux_ppc_17.0_uclibc_hardened/package.mask
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.mask	2020-01-24 21:40:38.733390292 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/package.mask	2020-01-24 21:40:38.850044259 -0800
@@ -2,19 +2,19 @@
 <dev-util/nvidia-cuda-sdk-9
 <dev-util/nvidia-cuda-toolkit-9
 <sys-devel/gcc-6
-=virtual/libudev-232-r1
->=www-client/firefox-54.0
+<sys-libs/glibc-2.25
 >virtual/libelf-2
+app-crypt/hmaccalc
 dev-java/ecj-gcj
 dev-java/gcj-jdk
-dev-lang/rust-bin
 dev-libs/elfutils
+net-fs/openafs-kernel
 sys-apps/systemd
+sys-devel/prelink
 sys-fs/udev
 sys-libs/glibc
 sys-libs/musl
 sys-libs/pam
-sys-libs/uclibc-ng
 ~virtual/perl-Archive-Tar-2.240.0
 ~virtual/perl-Attribute-Handlers-0.990.0
 ~virtual/perl-B-Debug-1.240.0
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/packages default_linux_ppc_17.0_uclibc_hardened/packages
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/packages	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/packages	2020-01-24 21:40:38.843378318 -0800
@@ -15,6 +15,7 @@
 *sys-apps/busybox
 *sys-apps/coreutils
 *sys-apps/diffutils
+*sys-apps/elfix
 *sys-apps/file
 *sys-apps/gawk
 *sys-apps/grep
@@ -23,7 +24,6 @@
 *sys-apps/less
 -*sys-apps/man-pages
 *sys-apps/net-tools
-*sys-apps/sandbox
 *sys-apps/sed
 *sys-apps/util-linux
 *sys-apps/which
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/packages.build default_linux_ppc_17.0_uclibc_hardened/packages.build
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/packages.build	2020-01-24 21:40:38.733390292 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/packages.build	2020-01-24 21:40:38.850044259 -0800
@@ -5,12 +5,9 @@
 app-shells/bash:0
 dev-libs/libiconv
 dev-libs/libintl
-dev-python/pyxattr
-dev-util/pkgconf
 dev-util/pkgconfig
 net-misc/rsync
 net-misc/wget
-sys-apps/attr
 sys-apps/baselayout
 sys-apps/coreutils
 sys-apps/diffutils
@@ -22,7 +19,6 @@
 sys-apps/makedev
 sys-apps/net-tools
 sys-apps/portage
-sys-apps/sandbox
 sys-apps/sed
 sys-apps/which
 sys-devel/autoconf
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use default_linux_ppc_17.0_uclibc_hardened/package.use
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/package.use	2020-01-24 21:40:38.846711289 -0800
@@ -1,6 +1,7 @@
 =dev-lang/python-2* -berkdb
 app-admin/sudo -ldap
 app-emulation/qemu qemu_softmmu_targets_ppc
+dev-lang/python gdbm
 dev-ml/llvm-ocaml llvm_targets_BPF
 dev-util/pkgconfig internal-glib
 dev-vcs/git-bz python_single_target_python2_7 python_targets_python2_7
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use.force default_linux_ppc_17.0_uclibc_hardened/package.use.force
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use.force	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/package.use.force	2020-01-24 21:40:38.846711289 -0800
@@ -1,9 +1,7 @@
 >=dev-games/simgear-2016.2.1 dns
->=mail-client/thunderbird-45.0 system-libevent
 >=net-fs/netatalk-3.1.3 acl
->=www-client/firefox-45.0 system-libevent
->=www-client/seamonkey-2.40 system-libevent
 >gnome-base/nautilus-3.29 seccomp
+app-arch/tar xattr
 app-emulation/wine-staging staging
 app-misc/tracker-miners seccomp
 app-shells/bash:0 readline
@@ -25,7 +23,8 @@
 net-libs/webkit-gtk seccomp
 net-misc/netkit-bootparamd libtirpc
 sci-astronomy/esomidas libtirpc
-sys-apps/portage ipc
+sys-apps/coreutils xattr
+sys-apps/portage ipc xattr
 sys-boot/refind gnuefi
 sys-cluster/glusterfs libtirpc
 sys-devel/clang llvm_targets_PowerPC
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use.mask default_linux_ppc_17.0_uclibc_hardened/package.use.mask
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/package.use.mask	2020-01-24 21:40:38.733390292 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/package.use.mask	2020-01-24 21:40:38.846711289 -0800
@@ -73,7 +73,6 @@
 app-office/libreoffice libreoffice_extensions_wiki-publisher
 app-office/lyx subversion
 app-portage/g-sorcery bson
-app-shells/bash mem-scramble
 app-shells/mksh -static
 app-text/asciidoc test
 app-text/ghostscript-gpl l10n_ja
@@ -139,7 +138,7 @@
 dev-util/catalyst system-bootloader
 dev-util/geany-plugins git gtkspell
 dev-util/intel-ocl-sdk system-clang
-dev-vcs/git mediawiki mediawiki-experimental
+dev-vcs/git mediawiki mediawiki-experimental pcre-jit
 dev-vcs/subversion kwallet
 games-arcade/savagewheels fmod
 games-board/stockfish armv7
@@ -158,7 +157,7 @@
 kde-misc/kdeconnect pulseaudio
 mail-client/claws-mail gtk3 litehtml pda
 mail-client/mutt notmuch
-mail-client/thunderbird lto rust system-cairo
+mail-client/thunderbird rust system-cairo
 mate-base/mate bluetooth
 media-gfx/exact-image php
 media-gfx/imagemagick heif
@@ -200,6 +199,7 @@
 net-analyzer/wireshark bcg729
 net-dns/pdnsd -isdn -urandom
 net-dns/unbound dnstap
+net-fs/openafs modules
 net-fs/samba ceph dmapi fam system-heimdal
 net-im/choqok telepathy
 net-im/pidgin zephyr
@@ -210,11 +210,9 @@
 net-libs/gnutls dane test-full
 net-libs/libproxy kde
 net-libs/ngtcp2 ssl
-net-libs/pjproject g729
 net-misc/bfgminer adl lm-sensors
 net-misc/connman ofono openconnect
 net-misc/curl nghttp3 quiche
-net-misc/iputils idn
 net-misc/modemmanager qmi
 net-misc/openssh pie
 net-misc/spice-gtk smartcard
@@ -258,18 +256,16 @@
 sys-cluster/openmpi openmpi_fabrics_dapl openmpi_fabrics_knem openmpi_fabrics_ofed openmpi_fabrics_psm openmpi_ofed_features_connectx-xrc openmpi_ofed_features_control-hdr-padding openmpi_ofed_features_dynamic-sl openmpi_ofed_features_failover openmpi_ofed_features_rdmacm openmpi_ofed_features_udcm openmpi_rm_slurm
 sys-devel/clang default-libcxx
 sys-devel/clang-runtime crt libcxx openmp sanitize
-sys-devel/gcc awt hardened jit libssp pie sanitize vtv
-sys-devel/gettext nls
+sys-devel/gcc awt -hardened jit libssp pie sanitize vtv
 sys-devel/llvm gold
 sys-fabric/ofed ofed_drivers_ehca
 sys-firmware/seabios -debug
 sys-fs/dmraid dietlibc
-sys-fs/e2fsprogs nls
 sys-fs/multipath-tools rbd
 sys-fs/xfsdump nls
 sys-fs/zfs kernel-builtin
 sys-kernel/linux-firmware initramfs unknown-license
-sys-libs/glibc cet hardened vanilla
+sys-libs/glibc cet -hardened vanilla
 sys-libs/gwenhywfar fox
 sys-libs/libcxx libcxxrt
 sys-libs/libxcrypt system
@@ -279,7 +275,7 @@
 virtual/mysql embedded
 virtual/notification-daemon kde
 www-client/elinks guile
-www-client/firefox lto pgo rust system-cairo
+www-client/firefox gmp-autoupdate pgo rust system-cairo
 www-client/links suid
 www-client/otter webengine
 www-servers/nginx luajit nginx_modules_http_lua
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/use.force default_linux_ppc_17.0_uclibc_hardened/use.force
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/use.force	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/use.force	2020-01-24 21:40:38.843378318 -0800
@@ -1,10 +1,11 @@
 abi_ppc_32
 big-endian
 elibc_glibc
-elibc_musl
 elibc_uclibc
 kernel_linux
+pie
 ppc
 split-usr
+ssp
 uclibc
 userland_GNU
diff -u default_linux_powerpc_ppc32_17.0_uclibc_hardened/use.mask default_linux_ppc_17.0_uclibc_hardened/use.mask
--- default_linux_powerpc_ppc32_17.0_uclibc_hardened/use.mask	2020-01-24 21:40:38.730057322 -0800
+++ default_linux_ppc_17.0_uclibc_hardened/use.mask	2020-01-24 21:40:38.843378318 -0800
@@ -99,14 +99,15 @@
 elibc_bionic
 elibc_glibc
 elibc_mintlib
--elibc_musl
-elibc_uclibc
+elibc_musl
+-elibc_uclibc
 -fdk
 firebird
 fixed-point
 fortune
 gcl
 gnuefi
+-hardened
 hdf
 hppa
 ia64
@@ -172,6 +173,7 @@
 orc
 packagekit
 pam
+pch
 php_targets_php5-6
 php_targets_php7-0
 php_targets_php7-1
@@ -183,6 +185,7 @@
 prefix
 prefix-guest
 prefix-stack
+prelink
 prelude
 ps3
 python_single_target_pypy3
@@ -209,7 +212,6 @@
 texmacs
 -uclibc
 userland_BSD
-utmp
 vaapi
 vdpau
 video_cards_amdgpu
@@ -238,7 +240,6 @@
 video_cards_vmware
 vidix
 vis
-wtmp
 x86
 x86-fbsd
 xvmc
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland is now default/linux/ppc64/17.0
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland/make.defaults default_linux_ppc64_17.0/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland/make.defaults	2020-01-24 21:40:38.986696049 -0800
+++ default_linux_ppc64_17.0/make.defaults	2020-01-24 21:40:39.100017046 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="yes"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="acl berkdb bzip2 cli crypt cxx dri fortran gdbm ibm iconv ipv6 ncurses nls nptl openmp pam pcre readline seccomp split-usr ssl tcpd unicode xattr zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/desktop is now default/linux/ppc64/17.0/desktop
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_desktop/make.defaults default_linux_ppc64_17.0_desktop/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_desktop/make.defaults	2020-01-24 21:40:39.236668835 -0800
+++ default_linux_ppc64_17.0_desktop/make.defaults	2020-01-24 21:40:39.359988743 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="yes"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="X a52 aac acl acpi alsa berkdb bluetooth branding bzip2 cairo cdda cdr cli consolekit crypt cups cxx dbus dri dts dvd dvdr emboss encode exif flac fortran gdbm gif gpm gtk ibm iconv icu ipv6 jpeg lcms ldap libnotify mad mng mp3 mp4 mpeg ncurses nls nptl ogg opengl openmp pam pango pcre pdf png policykit ppds qt5 readline sdl seccomp spell split-usr ssl startup-notification svg tcpd tiff truetype udev udisks unicode upower usb vorbis wxwidgets x264 xattr xcb xml xv xvid zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/desktop/gnome is now default/linux/ppc64/17.0/desktop/gnome
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_desktop_gnome/make.defaults default_linux_ppc64_17.0_desktop_gnome/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_desktop_gnome/make.defaults	2020-01-24 21:40:39.509972415 -0800
+++ default_linux_ppc64_17.0_desktop_gnome/make.defaults	2020-01-24 21:40:39.633292323 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="yes"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="X a52 aac acl acpi alsa berkdb bluetooth branding bzip2 cairo cdda cdr cli colord -consolekit crypt cups cxx dbus dri dts dvd dvdr eds elogind emboss encode evo exif flac fortran gdbm gif gnome gnome-keyring gnome-online-accounts gpm gstreamer gtk ibm iconv icu introspection ipv6 jpeg lcms ldap libnotify libsecret mad mng mp3 mp4 mpeg nautilus ncurses networkmanager nls nptl ogg opengl openmp pam pango pcre pdf png policykit ppds pulseaudio qt5 readline sdl seccomp spell split-usr ssl startup-notification svg tcpd tiff tracker truetype udev udisks unicode upower usb vorbis wxwidgets x264 xattr xcb xml xv xvid zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/desktop/gnome/systemd is now default/linux/ppc64/17.0/desktop/gnome/systemd
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_desktop_gnome_systemd/make.defaults default_linux_ppc64_17.0_desktop_gnome_systemd/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_desktop_gnome_systemd/make.defaults	2020-01-24 21:40:39.783275995 -0800
+++ default_linux_ppc64_17.0_desktop_gnome_systemd/make.defaults	2020-01-24 21:40:39.906595903 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="yes"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="X a52 aac acl acpi alsa berkdb bluetooth branding bzip2 cairo cdda cdr cli colord -consolekit crypt cups cxx dbus dri dts dvd dvdr eds elogind emboss encode evo exif flac fortran gdbm gif gnome gnome-keyring gnome-online-accounts gpm gstreamer gtk ibm iconv icu introspection ipv6 jpeg lcms ldap libnotify libsecret mad mng mp3 mp4 mpeg nautilus ncurses networkmanager nls nptl ogg opengl openmp pam pango pcre pdf png policykit ppds pulseaudio qt5 readline sdl seccomp spell split-usr ssl startup-notification svg systemd tcpd tiff tracker truetype udev udisks unicode upower usb vorbis wxwidgets x264 xattr xcb xml xv xvid zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/developer is now default/linux/ppc64/17.0/developer
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_developer/make.defaults default_linux_ppc64_17.0_developer/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_developer/make.defaults	2020-01-24 21:40:40.049913634 -0800
+++ default_linux_ppc64_17.0_developer/make.defaults	2020-01-24 21:40:40.163234630 -0800
@@ -45,12 +45,11 @@
 PORTAGE_ELOG_CLASSES=" qa"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="yes"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="X a52 aac acl acpi alsa berkdb bluetooth branding bzip2 cairo cdda cdr cli consolekit crypt cups cxx dbus dri dts dvd dvdr emboss encode exif flac fortran gdbm gif gpm gtk ibm iconv icu ipv6 jpeg lcms ldap libnotify mad mng mp3 mp4 mpeg ncurses nls nptl ogg opengl openmp pam pango pcre pdf png policykit ppds qt5 readline sdl seccomp snmp spell split-usr ssl startup-notification svg tcpd tiff truetype udev udisks unicode upower usb vorbis wxwidgets x264 xattr xcb xml xv xvid zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/little-endian is now default/linux/ppc64le/17.0
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_little-endian/make.defaults default_linux_ppc64le_17.0/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_little-endian/make.defaults	2020-01-24 21:40:40.313218302 -0800
+++ default_linux_ppc64le_17.0/make.defaults	2020-01-24 21:40:40.446537122 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="no"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="acl berkdb bzip2 cli crypt cxx dri fortran gdbm ibm iconv ipv6 ncurses nls nptl openmp pam pcre readline seccomp split-usr ssl tcpd unicode xattr zlib"
********************
####################
default/linux/powerpc/ppc64/17.0/64bit-userland/little-endian/systemd is now default/linux/ppc64le/17.0/systemd
diff -u default_linux_powerpc_ppc64_17.0_64bit-userland_little-endian_systemd/make.defaults default_linux_ppc64le_17.0_systemd/make.defaults
--- default_linux_powerpc_ppc64_17.0_64bit-userland_little-endian_systemd/make.defaults	2020-01-24 21:40:40.599853764 -0800
+++ default_linux_ppc64le_17.0_systemd/make.defaults	2020-01-24 21:40:40.723173672 -0800
@@ -42,12 +42,11 @@
 PHP_TARGETS="php7-2"
 POSTGRES_TARGETS="postgres10 postgres11"
 PROFILE_ARCH="ppc64"
-PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
+PROFILE_ONLY_VARIABLES=" ELIBC IUSE_IMPLICIT KERNEL USERLAND USE_EXPAND_IMPLICIT USE_EXPAND_UNPREFIXED USE_EXPAND_VALUES_ELIBC USE_EXPAND_VALUES_KERNEL USE_EXPAND_VALUES_USERLAND ARCH USE_EXPAND_VALUES_ARCH ARCH USE_EXPAND_VALUES_ARCH"
 PYTHONDONTWRITEBYTECODE="1"
 PYTHON_SINGLE_TARGET="python3_6"
 PYTHON_TARGETS="python2_7 python3_6"
 RUBY_TARGETS="ruby24 -ruby25"
-SYMLINK_LIB="no"
 TWISTED_DISABLE_WRITING_OF_PLUGIN_CACHE="1"
 UNINSTALL_IGNORE="/lib/modules/* /var/run /var/lock"
 USE="acl berkdb bzip2 cli crypt cxx dri fortran gdbm ibm iconv ipv6 ncurses nls nptl openmp pam pcre readline seccomp split-usr ssl systemd tcpd udev unicode xattr zlib"
********************

```

gyakovlev commented:
test is here

https://github.com/gyakovlev/profile-cleanup/blob/ppc-profile-test/ppc-profile-test.sh

gyakovlev commented:
last open question is " what do we do with SYMLINK_LIB in ppc64 profiles?
ppc64-little-endian had it set explicitly, if it's unset it's same as 'no' afaik.
we ca use  this profile change to finally migrate from symlinked lib on ppc64-big-endian.
`unsymlink-lib` is battle-tested and I can re-use news item from amd64 17.1 profiles.

mattst88 commented:
Yes, now would be a great opportunity to switch SYMLINK_LIB to no (see bug https://bugs.gentoo.org/640184)

gyakovlev commented:
merged.
